code/game/objects/structures/stool_bed_chair_nest/chairs.dm

```diff
@@ -132,4 +132,96 @@
 	icon_state = "officechair_white"
 
 /obj/structure/stool/bed/chair/office/dark
-	icon_state = "officechair_dark"
\ No newline at end of file
+	icon_state = "officechair_dark"
+
+/obj/structure/stool/bed/chair/office/red
+	icon_state = "officechair_red"
+
+/obj/structure/stool/bed/chair/office/blue
+	icon_state = "officechair_blue"
+
+//chairs from urist mcstation, credit to them
+
+/obj/structure/stool/bed/chair/couch
+	name = "couch"
+	desc = "A couch. Looks pretty comfortable."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "chair"
+	color = rgb(255,255,255)
+	var/image/armrest = null
+	var/couchpart = 0 //0 = middle, 1 = left, 2 = right
+
+/obj/structure/stool/bed/chair/couch/New()
+	if(couchpart == 1)
+		armrest = image("icons/obj/objects.dmi", "armrest_left")
+		armrest.layer = MOB_LAYER + 0.1
+	else if(couchpart == 2)
+		armrest = image("icons/obj/objects.dmi", "armrest_right")
+		armrest.layer = MOB_LAYER + 0.1
+
+	return ..()
+
+/obj/structure/stool/bed/chair/couch/post_buckle_mob(mob/living/M)
+	if(buckled_mob)
+		overlays += armrest
+	else
+		overlays -= armrest
+
+/obj/structure/stool/bed/chair/couch/left
+	couchpart = 1
+	icon_state = "couch_left"
+
+/obj/structure/stool/bed/chair/couch/right
+	couchpart = 2
+	icon_state = "couch_right"
+
+/obj/structure/stool/bed/chair/couch/middle
+	icon_state = "couch_middle"
+
+/obj/structure/stool/bed/chair/couch/left/black
+	color = rgb(167,164,153)
+
+/obj/structure/stool/bed/chair/couch/right/black
+	color = rgb(167,164,153)
+
+/obj/structure/stool/bed/chair/couch/middle/black
+	color = rgb(167,164,153)
+
+/obj/structure/stool/bed/chair/couch/left/teal
+	color = rgb(0,255,255)
+
+/obj/structure/stool/bed/chair/couch/right/teal
+	color = rgb(0,255,255)
+
+/obj/structure/stool/bed/chair/couch/middle/teal
+	color = rgb(0,255,255)
+
+/obj/structure/stool/bed/chair/couch/left/beige
+	color = rgb(255,253,195)
+
+/obj/structure/stool/bed/chair/couch/right/beige
+	color = rgb(255,253,195)
+
+/obj/structure/stool/bed/chair/couch/middle/beige
+	color = rgb(255,253,195)
+
+/obj/structure/stool/bed/chair/couch/left/brown
+	color = rgb(255,113,0)
+
+/obj/structure/stool/bed/chair/couch/right/brown
+	color = rgb(255,113,0)
+
+/obj/structure/stool/bed/chair/couch/middle/brown
+	color = rgb(255,113,0)
+
+
+/obj/structure/stool/bed/chair/couch/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/weapon/wrench))
+		playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
+		new /obj/item/stack/sheet/metal(src.loc)
+		del(src)
+	if(istype(W, /obj/item/weapon/chair_painter))
+		var/obj/item/weapon/chair_painter/C = W
+		color = rgb(C.red,C.green,C.blue)

```

Comment:
the parent type plays a different sound and drops a different item, so i had to clarify.
HOWEVER, i just noticed that i used del instead of qdel, so that needs to be fixed